### DISCUSSION TITLE
Wordstar key bindings implemented beautifully

### DISCUSSION THREAD

**Discussion by berwynhoyt** [xremap/xremap#589] [Ideas]
Hi Takashi.

I have used xremap to implement [Wordstar key mappings for Linux here](https://github.com/berwynhoyt/wordstar-keys). I've been looking for a very long time for a Linux tool like yours, and I was so pleased to find it. This is just a note to suggest you may wish to link this mapping for the use of others.

By way of comparison, it took me only one day to implement these using xremap, whereas it took me a week, full-time, to implent wordstar key mappings as an xkb keyboard map. It was a bit of a feat to get them to work with xkb at all as xkb isn't set up for the purpose. I've been using those for some time for lack of anything better. But, although they work reasonably, yours xremap is far better for multi-key output and application specific overrides.

**Comment by berwynhoyt** [comment]
I meant to ask: **would you consider a pull request** if I made one that adds an xremap installer like the one I have created in the [Wordstar key mappings repository](https://github.com/berwynhoyt/wordstar-keys). It automatically sets up uinput permissions, downloads the latest binary, and installs it as a service.

The main gotcha with the installer is that it asks the user which of the 6 binaries to download. Ideally, it would automatically detect based on the user's platform. This would be easy to add for architecture, but I'm not sure how to detect sway, X11 or gnome. I use only one of them (Cinnamon/Mint).

As an aside, I'm sure you've considered building just one binary for sway/X11/gnome that simply uses a command line switch to select the window detection mechanism. This would be attractive. What was your reason for not doing so?

**Comment by k0kubun** [comment]
I think xremap should be a single binary that supports all platforms and automatically detects which integration to use, allowing an option to override it. Does that make sense?

**Reply by berwynhoyt** [reply]
Thanks. You seem to be agreeing with me that you'd like to have a single binary. Yes, if it can auto-detect, so much the better. I'm curious why you decided not to implement it this way in the first place.
